src/compas_fab/robots/robot.py

```diff
@@ -1014,6 +1068,114 @@ def inverse_kinematics(self, frame_WCF, start_configuration=None,
 
         return configuration.scaled(self.scale_factor)
 
+    def iter_inverse_kinematics(self,
```

Comment:
Here, one issue is that `iter_*` functions should return iterators, not the blocking behavior implemented here.

Reply:
This function, as it stands right now, it's not composable because it calls `position_constraint_from_max_reach` internally, which, as @romanarust points out, it's specific to your problem. But if you shuffle things around a little bit, and it will become much more usable.

A quick draft of what I would have expected of this function is this:
```python
def iter_inverse_kinematics(self, ....., max_iterations):
   for i in range(max_iterations):
      yield self.inverse_kinematics(...)
```

This would call ik for a max of the `max_iterations` parameter, but can also be used as a proper iterator, but breaking at the first acceptable result from the calling side will stop the calls from occurring (instead of having to wait for all the iterations to finish). That puts the calling code in control.

Obviously calling repeatedly IK only makes sense in some cases (redundant robot, or random seed).